src/requests/create-signature.ts

```diff
@@ -0,0 +1,87 @@
+import * as crypto from 'crypto'
+import type { CanonicalRequest, Secret, Timestamp } from './typings'
+import { CanonicalRequestValidator, SecretValidator, TimestampValidator } from './typings'
+import { getNormalizedEncodedURI, getNormalizedHeaders } from './utils'
+
+const hash = (normalizedCanonicalRequest: CanonicalRequest, secret: string) => {
+  const stringifiedHeaders = Object.entries(normalizedCanonicalRequest.headers!)
+    .map(([key, value]) => `${key}:${value}`)
+    .join(';')
+
+  const stringifiedRequest = [
+    normalizedCanonicalRequest.method,
+    normalizedCanonicalRequest.path,
+    stringifiedHeaders,
+    normalizedCanonicalRequest.body,
+  ].join('\n')
+
+  const hmac = crypto.createHmac('sha256', secret)
+
+  hmac.update(stringifiedRequest)
+
+  return hmac.digest('hex')
+}
+
+/**
+ * Given a secret, a canonical request and a timestamp, generates a signature.
+ * It can be used to verify canonical requests to assess authenticity of the
+ * sender and integrity of the payload.
+ *
+ * ~~~
+ * const {createSignature} = require('contentful-node-apps-toolkit')
+ * const {omit} = require('lodash')
+ * const {server} = require('./imaginary-server')
+ *
+ * const SECRET = process.env.SECRET
+ *
+ * server.post('/api/my-resources', (req, res) => {
+ *   const incomingSignature = req.headers['x-signature']
+ *   const incomingTimestamp = req.headers['x-timestamp']
+ *   const now = Date.now()
+ *
+ *   if (!incomingSignature) {
+ *     res.send(400, 'Missing signature')
+ *   }
+ *
+ *   if (now - incomingTimestamp > 1000) {
+ *     res.send(408, 'Request too old')
+ *   }
+ *
+ *   const computedSignature = createSignature(
+ *     SECRET,
+ *     {
+ *       method: req.method,
+ *       path: req.url
+ *       headers: omit(req.headers, 'x-signature'),
```

Comment:
I would expect it to get all headers. In the proposal we said that signed headers will be listed as a value of the `x-contentful-signed-headers` header

Reply:
As in [this orther comment](https://github.com/contentful/node-apps-toolkit/pull/11#discussion_r512051450) the idea was to showcase a generic flow. How to use it within Contentful ecosystem (for example, with signed requests) it should be documented elsewhere in my eyes. WDYT?